[ds-heap] binary heap implementation

pathfinder177 opened the pull request:
* adding binary heap

Eugen-Kuzovlev committed:
[ds-heap] added heap _create, _erase_value functions

Yanpas commented on heap.c:
```diff
@@ -0,0 +1,99 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include "node.h"
```
Do you need a node?

pathfinder177 replied:
Deleted

Yanpas reviewed (dismissed):
I suggest you to review the API behind the heap. Most of the times we use heap to get "the largest (smallest) element: e.g. the timers queue. In this sense the heap behaves like a queue with push, pop and peek operations

Eugen-Kuzovlev committed:
[ds-heap] began to change interface

pathfinder177 commented:
> I suggest you to review the API behind the heap. Most of the times we use heap to get "the largest (smallest) element: e.g. the timers queue. In this sense the heap behaves like a queue with push, pop and peek operations

Began to implement

Eugen-Kuzovlev committed:
[ds-heap] implemented heap_push

[ds-heap] implemented heap_peek

[ds-heap] implemented heap_pop, deleted needless heap.size field
